### PR TITLE
Upgrade backend to echo v4

### PR DESCRIPTION
**Pull request by loleg**
Updating the echo framework is a major upgrade from v3 to v4.

See details in #216

### CODE REVIEW

**Comment by loleg** [comment]
Not building due to this issue at the moment:

```
go: downloading golang.org/x/sys v0.13.0
go install github.com/wadey/gocovmerge
no required module provides package github.com/wadey/gocovmerge; to add it:
        go get github.com/wadey/gocovmerge
make: *** [Makefile:84: init] Error 1
```

**Comment by loleg** [comment]
Quite sure my issue has to do with this:

https://github.com/ProxeusApp/proxeus-core/blob/37991fdd57d76d4c517fba7cfadcdc8a7adc5550/main/www/initial.go#L40

**Comment by loleg** [comment]
All tests passing. I'm doing some additional local validation.

**Review by tafonina — APPROVED** [review]
Looks good, the only thing - be sure that the data is read from env file without manual export as was specified in the build_all.md file

File: docs/build_all.md [hunk 1]
```diff
@@ -54,21 +53,12 @@ cd proxeus-core
...
-```
-export PROXEUS_DATA_DIR=./data
-export PROXEUS_INFURA_API_KEY=<Your Infura project ID>
-export PROXEUS_SPARKPOST_API_KEY=<Your SparkPost API Key>
-export PROXEUS_ENCRYPTION_SECRET_KEY=<A 32-character random string>
-export PROXEUS_BLOCKCHAIN_CONTRACT_ADDRESS=0x1d3e5c81bf4bc60d41a8fbbb3d1bae6f03a75f71
-export PROXEUS_ALLOW_HTTP=true
-export PROXEUS_PLATFORM_DOMAIN=localhost
-export PROXEUS_EMAIL_FROM=youremail@example.com
+Before building Proxeus, make sure to set all [required environment variables](build_docker.md) (i.e. with an `.env` file).
```
Comment (loleg) [inline docs/build_all.md:59]: As @tafonina pointed out, this is an important instruction. Make it more explicit, e.g. rename the section in the build_docker documentation.

Comment (loleg) [reply docs/build_all.md:59]: I'll make a note of this request in #452

**Pull request merged**